### ISSUE TITLE
Are `usize` and `i64` not supported?

### ISSUE THREAD

**Issue by affanshahid** [1Password/typeshare#24]
Ran typeshare on my project and got the following errors:

```
thread '<unnamed>' panicked at 'failed to parse a rust type: ["usize"]', /Users/affan/.cargo/registry/src/github.com-1ecc6299db9ec823/typeshare-cli-1.0.0/src/main.rs:209:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'failed to parse a rust type: ["i64"]', /Users/affan/.cargo/registry/src/github.com-1ecc6299db9ec823/typeshare-cli-1.0.0/src/main.rs:209:17
```

i64 can be represented using a JS BigInt.

**Comment by jeremychone** [comment]
as with `ts-rs` would be nice to have a `#[typeshare(type = "number")]` property attribute.

**Comment by InquisitivePenguin** [comment]
Thanks for opening this issue. I'll look into whether this is possible.

**Comment by BenJeau** [comment]
Getting a similar error with `u64` and `isize`, those four types seems to be "disabled" here

https://github.com/1Password/typeshare/blob/fbf5aea145b7895f3a998db050bd972a9af79e05/core/src/rust_types.rs#L225-L227

But the types exist in the [SpecialRustType](https://github.com/1Password/typeshare/blob/fbf5aea145b7895f3a998db050bd972a9af79e05/core/src/rust_types.rs#L108) and looking in the four supported languages, they all have mappings for those four types.

Locally I've changed those three lines in `rust_types.rs` to parse the types and I'm not getting any errors, but I'm guessing that there must be a reason that those lines/errors exist? It works for go, swift, typescript, and kotlin.

### Changes

In `rust_types.rs`:

```diff
-                    "u64" | "i64" | "usize" | "isize" => {
-                        return Err(RustTypeParseError::UnsupportedType(vec![id]))
-                    }
+                    "u64" => RustType::Special(SpecialRustType::U64),
+                    "usize" => RustType::Special(SpecialRustType::USize),
+                    "i64" => RustType::Special(SpecialRustType::I64),
+                    "isize" => RustType::Special(SpecialRustType::ISize),
```

and in `typescript.rs`:

```diff
            SpecialRustType::U64
            | SpecialRustType::I64
            | SpecialRustType::ISize
            | SpecialRustType::USize => {
-                panic!("64 bit types not allowed in Typeshare")
+                Ok("number".into())
            }
```

**Comment by Lucretiel** [comment]
I think part of the problem is that JSON is usually going to be the medium of exchange for typeshared data, and so when you're loading into javascript / typescript you'd need to *not* use `JSON.parse`, since it doesn't care what type you intended and will just load into a `number`. A lot of the complexity around typeshare enums is similarly related to conforming to the default behavior of various languages when dealing with their equivalent of enums.

**Comment by BenJeau** [comment]
I understand that most people would use JSON (as it is one of the most popular transport medium), but it is not the _only_ transport medium in which `typeshare` types can be used, e.g. FFIs

If users of `typeshare` decide to use it with 64bit types, maybe the library should just give a warning and maybe suggest an alternative library instead of the builtin `JSON.parse`/`JSON.stringify`, such as https://github.com/blitz-js/superjson, for the TypeScript/JavaScript side? It seems like this feature is blocked on the fact that JavaScript does not play well by default with 64bit values in JSON, whilst other languages are fine.

**Comment by mattyg** [comment]
I just ran into this -- I also need to serialize u64 types. Seems like an oversight to always panic when encountering these types as there are types in the destination languages to support large numbers. At least let the developer set the type themselves.

**Comment by AlexanderProd** [comment]
@mattyg A workaround for this is to use `#[typeshare(serialized_as = "number")]`.

**Comment by Uzaaft** [comment]
> @mattyg A workaround for this is to use `#[typeshare(serialized_as = "number")]`.

This workaround doesn't work. Tested stable and the beta branch.

**Comment by tomjw64** [comment]
@LuminaSapphira [I made a PR](https://github.com/1Password/typeshare/pull/140) for supporting this a few months back. What do you think of that approach (i.e. allow only if the output type is explicitly specified?)

**Comment by tomjw64** [comment]
I resolved the existing merge conflicts in the above PR.  I would still really appreciate a review + workflow approval from a maintainer!

**Comment by mjovanc** [comment]
Any news on this?
